### PR TITLE
Integration specs fix n+1

### PR DESCRIPTION
**Pull request by edgardsoza**
Make sure that the N+1 problem is solved when fetching all posts and their comments for a user. Done
Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh). Done
In your PR description include the screenshots of your app's logs before and after the fix. Done
Before
comments before N+1 solution
posts before N+1 solution

After:
posts after N+1 solution
comments after N+1 solution

Integration specs
Use Capybara to write integration tests for each view in your project.

Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

User index page:

I can see the username of all other users.
I can see the profile picture for each user.
I can see the number of posts each user has written.
When I click on a user, I am redirected to that user's show page.
user show page: Done

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see the user's bio.
I can see the user's first 3 posts.
I can see a button that lets me view all of a user's posts.
When I click a user's post, it redirects me to that post's show page.
When I click to see all posts, it redirects me to the user's post's index page.
User post index page: Done

I can see the user's profile picture.
I can see the user's username.
I can see the number of posts the user has written.
I can see a post's title.
I can see some of the post's body.
I can see the first comments on a post.
I can see how many comments a post has.
I can see how many likes a post has.
I can see a section for pagination if there are more posts than fit on the view.
When I click on a post, it redirects me to that post's show page.
Post show page: Done

I can see the post's title.
I can see who wrote the post.
I can see how many comments it has.
I can see how many likes it has.
I can see the post body.
I can see the username of each commentor.
I can see the comment each commentor left.
Be sure to include integration specs for any other features you may have implemented! 
Done

### CODE REVIEW

File: spec/views/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    it 'renders 3 recent posts of the user' do
+      expect(page).to have_content('Post #1')
+      expect(page).to have_content('Post #2')
+      expect(page).to have_content('Post #3')
+    end
+
+    it 'renders a button to redirect to posts page' do
+      expect(page).to have_content('See all posts')
+    end
+
```
Comment (Owusu-Desmond) [inline spec/views/user_show_spec.rb:42]: - [ ] You did a good job writing most test cases for the user show page :+1:  However, you did not write a test case that when you click on a user's post, it redirects to the post's show page so make sure you test this :pray:

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      expect(page).to have_content(@post1.likes_counter)
+      expect(page).to have_content(@post2.likes_counter)
+      expect(page).to have_content(@post3.likes_counter)
+    end
+
+    it 'renders pagination section if posts are more than 3' do
+      expect(page).to have_content('See more')
+    end
+  end
+end
```
Comment (Owusu-Desmond) [inline spec/views/post_index_spec.rb:79]: - [ ] You did a good job writing most test cases for the user post index page :+1:  However, you did not write a test case that when you click on a user's post, it redirects to the post's show page so make sure you test this :pray:

**Review by Owusu-Desmond — CHANGES_REQUESTED** [review]
Hi @edgardsoza and @AyomipoMajek , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
📌 Linters are passing ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DuaneDave — APPROVED** [review]
Hi @edgardsoza 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Good job writing all the required test cases

## Optional suggestions
N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**